lib/chef/provider/remote_directory.rb

```diff
@@ -67,13 +67,10 @@ def ls(path)
                  ::File::FNM_DOTMATCH)
 
         # Remove current directory and previous directory
-        files.reject! do |name|
+        files = files.reject do |name|
```

Comment:
I've put the `cleanpath` line back in so it needs to save the output for .reject to file.